homeassistant/components/sonos/diagnostics.py

```diff
@@ -112,7 +111,9 @@ async def async_generate_speaker_info(
     """Generate the diagnostic payload for a specific speaker."""
     payload = {}
```

Comment:
```suggestion
    payload: dict[str, Any] = {}
```
With that the `type: ignore` shouldn't be necessary anymore.
`dict[str, Any]` can also be added in `async_generate_media_info` above.